### PR TITLE
Selecting class

### PR DESCRIPTION
**Pull request by eperrio1**

### CODE REVIEW

File: src/main/java/AttendanceTracker/App.java [hunk 1]
```diff
@@ -10,5 +13,44 @@ public String getGreeting() {
 
     public static void main(String[] args) {
         System.out.println(new App().getGreeting());
+
+        // Will be done in different task currently in icebox
+        List<String> fall = new ArrayList<>();
+        fall.add("CPSC 1724");
+        fall.add("CPSC 2720");
+        fall.add("CPSC 3140");
+        System.out.println("\nCourses offered in the Fall");
```
Comment (WhonixJosh) [inline src/main/java/AttendanceTracker/App.java:22]: Move this to a function

File: src/main/java/AttendanceTracker/App.java [hunk 2]
```diff
@@ -10,5 +13,44 @@ public String getGreeting() {
...
+        String selectedClass = selectingClass(fall);
+        System.out.println(selectedClass);
+
+    
+    }
+
+
+       public static String selectingClass(List classes) {
+        Scanner myObj = new Scanner(System.in);
+        System.out.println("\nSelect a course: ");
```
Comment (WhonixJosh) [inline src/main/java/AttendanceTracker/App.java]: this needs to handle input mismatch exception if I input a non-string

Comment (WhonixJosh) [reply src/main/java/AttendanceTracker/App.java]: Move this to separate class outside of the App in main.java

File: src/test/java/AttendanceTracker/AppTest.java [hunk 3]
```diff
@@ -6,9 +6,34 @@
...
+    @Test void testselectingClass() {
+        List<String> mockList = new ArrayList<>();
+        mockList.add("MATH 1234");
+        mockList.add("MATH 5678");
+        mockList.add("MATH 3579");
+
+        String inputCourse = "MATH 1234";
+        InputStream in = new ByteArrayInputStream(inputCourse.getBytes());
+        System.setIn(in);
+        assertEquals("MATH 1234", App.selectingClass(mockList));
```
Comment (WhonixJosh) [inline src/test/java/AttendanceTracker/AppTest.java]: add a failing test

File: src/test/java/AttendanceTracker/AppTest.java [hunk 4]
```diff
@@ -6,9 +6,34 @@
...
+
+
 class AppTest {
     @Test void appHasAGreeting() {
         App classUnderTest = new App();
         assertNotNull(classUnderTest.getGreeting(), "app should have a greeting");
     }
+
+    
+    @Test void testselectingClass() {
```
Comment (WhonixJosh) [inline src/test/java/AttendanceTracker/AppTest.java]: Refactor the tested methods into a new class with the tests in a separate test file

File: src/main/java/AttendanceTracker/App.java [hunk 5]
```diff
@@ -10,5 +13,44 @@ public String getGreeting() {
...
+        String course = myObj.nextLine();
+
+        while(classes.contains(course) == false){
+            myObj = new Scanner(System.in);
```
Comment (WhonixJosh) [inline src/main/java/AttendanceTracker/App.java]: Don't forget to close the scanner, wrap this in a try{}finally{scanner.close()}

**Review by WhonixJosh — CHANGES_REQUESTED** [review]

File: src/main/java/AttendanceTracker/App.java (hunk 1 above)
Comment (eperrio1) [reply src/main/java/AttendanceTracker/App.java:22]: This is only temporary to see if my code work. The real function will be written once we do the "On the class view, the user can click the "Display List of Courses" button to see a list of every class for the given semester and year" user story.

**Pull request merged**